python/runtime/utils/PyRestRemoteClient.cpp

```diff
@@ -371,15 +383,29 @@ class PyNvcfRuntimeClient : public PyRestRemoteClient {
 
       if (m_logLevel > LogLevel::None)
         cudaq::log("Posting NVQC request now");
+      const auto requestStartTime = std::chrono::system_clock::now();
       auto resultJs =
           m_restClient.post(nvcfInvocationUrl(), "", requestJson, jobHeader,
                             /*enableLogging=*/false, /*enableSsl=*/true);
       cudaq::debug("Response: {}", resultJs.dump());
       while (resultJs.contains("status") &&
              resultJs["status"] == "pending-evaluation") {
         const std::string reqId = resultJs["reqId"];
+        const int elapsedTimeSecs =
+            std::chrono::duration_cast<std::chrono::seconds>(
+                std::chrono::system_clock::now() - requestStartTime)
+                .count();
         if (m_logLevel > LogLevel::None)
           cudaq::log("Polling NVQC result data for Request Id {}", reqId);
+        // Warns if the remaining time is less than this threshold.
+        constexpr int TIMEOUT_WARNING_SECS = 5 * 60; // 5 minutes.
+        const int remainingSecs = std::max(
+            m_availableFuncs[m_functionId].timeoutSecs - elapsedTimeSecs, 0);
+        if (remainingSecs < TIMEOUT_WARNING_SECS)
```

Comment:
@bmhowe23  I wonder whether we should log this "Approaching the wall time limit ..." line only once or repeat (like this).
Please let me know if you have any recommendations.

Reply:
> @bmhowe23 I wonder whether we should log this "Approaching the wall time limit ..." line only once or repeat (like this). Please let me know if you have any recommendations.

I put some suggestions in 69008e3f3.